### ISSUE TITLE
Setup a TravisCI instance to manage building documentation

### ISSUE THREAD

**Issue by OlegJakushkin** [kiliakis/BLonD-cpp#11]
It would be nice to have all that graphviz-Doxygen-html  documentation autogenerated on each new commit (or at least generated at all)

**Comment by OlegJakushkin** [comment]
Implemented in #12. Please pull in.

A demo can be viewed [here](http://olegjakushkin.github.io/BLonD-minimal-cpp/inherits.html)
Contains nice graphs covering:
- Class Hierarchy
- Call and caller graphs for functions
- Inheritance and Collaboration diagrams
Can be tweaked and expanded, allows markdown, latex documentation formatting

Requires gh-pages brunch to be created.

How to set up Doxygen autobuild into gh-pages brunch On Ubuntu 14.04
----
Create a [Personal Access Token](https://github.com/settings/tokens)
Only enable `public_repo` access for public repositories, "repo" for private.
Save the token somewhere as you can only see it once. ([reference](http://stackoverflow.com/a/33125422/1973207))

Install the travis gem:
```
sudo apt-get install ruby-dev
```
([reference](http://stackoverflow.com/a/8113213/1973207))
```
gem install travis
```

Then `cd` into your repository 
in case of merger please remove env->global->secure
and call:
```
travis encrypt GITHUB_API_KEY=<api-token> --add
```

also please update *repository* and *user* names in git push at the end of CI script.

**Comment by OlegJakushkin** [comment]
[On Ruby issue.pdf](https://github.com/kiliakis/BLonD-minimal-cpp/files/275717/On.Ruby.issue.pdf)
